### ISSUE TITLE
Do you consider Sass to be "real" programming?

### ISSUE THREAD

**Issue by esr360** [HugoGiraudel/ama#21]
Hi Hugo,

I've seen what Sass is capable of thanks to you. I've been having a debate with a friend who tells me it's not real programming, and to be fair all the evidence points towards him being right. But again, I've seen what it is capable of, especially with the more recent updates, and I would like to know your thoughts?

Thanks :)

**Comment by HugoGiraudel** [comment]
This is definitely a legit question. On one hand, Sass does borrow some concepts from programming languages, such as loops, functions and even some sort of inheritance through `@extend`. On the other hand, Sass is just a superset of CSS. The end goal is to generate stylesheets, not build softwares. Not build programs. Can we still talk about programming when it's not about developing programs? Not sure.

Sass definitely levelled up CSS in terms of complexity. While CSS is a fairly simple language that can be taught to basically anyone in just a few hours; Sass involves some features that makes the whole thing definitely harder to understand. Functions might seem obvious to us, but it's not that easy for someone with absolutely no programming background. Loops can also feel overwhelming at first.

Now, what [*I* did with Sass](#10) should not be taken as an example. The point was just to have fun and see how far I could push the limits of the language. So while building a JSON parser with nothing but Sass can be seen as programming (since it was actually about building a program encoding and decoding JSON), it is not a representative of the status and role of Sass as a language and tool.

Long story short: no, Sass is not programming. Because CSS is not either.
